{
  "modified": "2020-01-21T21:06:34Z",
  "published": "2013-08-18T01:00:00Z",
  "schema_version": "1.5.0",
  "id": "CVE-2013-4238",
  "aliases": [
    "PSF-2013-2"
  ],
  "details": "The ssl.match_hostname function in the SSL module in Python 2.6 through 3.4 does not properly handle a '\\0' character in a domain name in the Subject Alternative Name field of an X.509 certificate, which allows man-in-the-middle attackers to spoof arbitrary SSL servers via a crafted certificate issued by a legitimate Certification Authority, a related issue to CVE-2009-2408.",
  "affected": [
    {
      "package": {
        "ecosystem": "DHI",
        "name": "python",
        "purl": "pkg:dhi/python"
      },
      "ranges": [
        {
          "type": "SEMVER",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "2.6.9"
            },
            {
              "fixed": "3.10.0-alpha1"
            }
          ],
          "repo": "https://github.com/python/cpython"
        }
      ]
    }
  ],
  "references": [
    {
      "type": "REPORT",
      "url": "https://bugs.python.org/issue18709"
    }
  ]
}